operatorcourier/cli.py

```diff
@@ -0,0 +1,63 @@
+from operatorcourier import courier_api
+import argparse
+import sys
+
+def main():
+    """Generate the CLI bits
+    """
+    _CliParser()
+
+class _CliParser():
+    """Class that generates the command line bits for the operator-courier cli tool
+    """
+
+    def __init__(self):
+        # Generate the command level parser
+        parser = argparse.ArgumentParser(
+            description='Build, verify and push operator bundles into external app registry',
+            usage='''operator-courier <command> [<args>]
+
+These are the commands you can use:
+    verify      Create a bundle and test it for correctness.
+    push        Create a bundle, test it, and push it to an app registry.
+''')
+        parser.add_argument('command', help='Subcommand to run')
+
+        args = parser.parse_args(sys.argv[1:2])
+        if not hasattr(self, args.command):
```

Comment:
This can be done with subparsers https://docs.python.org/2/library/argparse.html#sub-commands

Reply:
I did read about those, but I thought the pattern presented here was very readable and didn't seem to come with any specific disadvantages. What is the value of using subparsers rather than the current pattern?

Reply:
I prefer to use stdlib solutions

Reply:
Isn't this part of the stdlib? https://docs.python.org/3/library/functions.html#getattr

If there's an obvious standard I'm not following, I can modify the source to follow it. But from my reading it wasn't clear that there was a pythonic standard to add commands to a cli tool. Is there some documentation you can point me to that shows using subcommand as a standard? 

I did take a look at some implementations and it seems like using subcommand here would prevent me from using the pattern here (creating separate methods for each subcommand) which to me is very readable.

Reply:
Subparsers are standard way how to add subcommands, but let leave this as it is now